cpp/test/Ice/operations/CMakeLists.txt

```diff
@@ -0,0 +1,320 @@
+if(NOT ${CMAKE_CURRENT_BINARY_DIR}/Ice)
```

Comment:
There are many common definitions in the test CMakeLists.txt, that apply to all Ice tests or all tests in general.
Is it possible to move these definitions to a common include file and not duplicate them in each test?

Reply:
I can take a look at that this evening, as well. The Ice/objects test does not build. It gets unresolved external symbol errors. The rest of them do. The uwp test I don't have much experience with but put it there in case a dev on your team wanted to experiment with it.

Reply:
I will come back to this when the slice dependencies are working.

Reply:
I think I'm making good progress on this. The functions seem to be working. I'm just fixing some typos and linking issues. There were two tests I didn't include::

`Glacier2/sessionControl`
`IceSSL/configuration`

Both these tests have unique include directories that don't work with the current functions, so they have their own unique CMakeLists. Once the build tests succeed I will push the changes and provide information on how to use the functions.

Reply:
I have pushed the test functions into a `test/cmake` directory. Each module directory has a CMakeLists to set up link dependencies and handle os specific variables, and each source directory has a CMakeLists to handle calling the function.

These cover targets that do not process an `ice` file:

```
CreateExecutableTest
CreateLibraryTest
```

These tests process a list of `ice` files:

```
CreateExecutableIceTest
CreateLibraryIceTest
```

These tests process a list of `ice` files that use the current source directory for the include path (`-I` setting):

```
CreateExecutableInternalIceTest
CreateLibraryInternalIceTest
```

`CreateAMDExecutableIceTest` is specifically for those targets that make use of the `ICE_AMD_TEST` compile definition.

`CreateExecutableImplIceTest` handles `ice` files that are generated with the implicit switch (`--impl-c++11`)

`CreateOpenSSLExecutableIceTest` uses the `ICE_USE_OPENSSL` compile definition. 

I was able to get the tests to successfully build with the default build options. There were some that had issues with external references or conversions in C++11 build. Testing on Debian WSL had some issues with SHA1 and SSL specifically. If you have some ideas about what I could be missing that would be excellent. Any other feedback is also greatly appreciated.